### ISSUE TITLE
[Windows] - Copy-Paste Crash with Hidden Channels

### ISSUE THREAD

**Issue by Toonlink8101** [BambooTracker/BambooTracker#393]
> #### Checklist
> 
> 
> 
> - [x] I am reporting exactly 1 bug with this issue.
> - [x] This bug hasn't already been reported.
> - [x] This bug hasn't already been fixed in the latest development build.

---

## Bug Description

Bambootracker crashes when copying and then pasting an order's content while certain channels are hidden.

## How to reproduce

1. Open a new module
2. Go into the view tab and hide any channel except for FM1 or ADPCM
3. Return to the note entry
4. Go into edit mode
4. Input ctrl+A twice to select all channels
5. Input copy (ctrl+C)
6. Input paste (ctrl+P)
7. Watch Bambootracker crash

## System Information

- **Operating System**:  Windows 10
- **BambooTracker Version**:  v0.4.6
- **Build Type**:  Official

**Comment by OPNA2608** [comment]
GDB Backtrace:

```
Thread 1 "BambooTracker" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff37b3580 (LWP 247417)]
PatternEditorPanel::pasteCopiedCells (this=0x39bb690, cursorPos=...) at gui/pattern_editor/pattern_editor_panel.cpp:1827
1827		PatternPosition pos = getPasteLeftAbovePosition(sCol, cursorPos, cells.front().size());
(gdb) bt
#0  PatternEditorPanel::pasteCopiedCells (this=0x39bb690, cursorPos=...) at gui/pattern_editor/pattern_editor_panel.cpp:1827
#1  0x00000000006577c1 in MainWindow::qt_static_metacall (_a=0x7fffffff9ae0, _id=11, _c=QMetaObject::InvokeMetaMethod, _o=0x80b290) at moc_mainwindow.cpp:574
#2  MainWindow::qt_metacall (this=0x80b290, _c=QMetaObject::InvokeMetaMethod, _id=11, _a=0x7fffffff9ae0) at moc_mainwindow.cpp:725
#3  0x00007ffff6fa4100 in ?? () from /nix/store/g6npkwh6m3rgl9hvc1j7vjg65jigrhc1-qtbase-5.15.2/lib/libQt5Core.so.5
#4  0x00007ffff7a3ae22 in QAction::triggered(bool) () from /nix/store/g6npkwh6m3rgl9hvc1j7vjg65jigrhc1-qtbase-5.15.2/lib/libQt5Widgets.so.5
#5  0x00007ffff7a3da68 in QAction::activate(QAction::ActionEvent) () from /nix/store/g6npkwh6m3rgl9hvc1j7vjg65jigrhc1-qtbase-5.15.2/lib/libQt5Widgets.so.5
#6  0x00007ffff7a3e647 in QAction::event(QEvent*) () from /nix/store/g6npkwh6m3rgl9hvc1j7vjg65jigrhc1-qtbase-5.15.2/lib/libQt5Widgets.so.5
#7  0x00007ffff7a4174f in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /nix/store/g6npkwh6m3rgl9hvc1j7vjg65jigrhc1-qtbase-5.15.2/lib/libQt5Widgets.so.5
#8  0x0000000000654bf3 in QApplicationWrapper::notify (this=<optimized out>, receiver=<optimized out>, event=<optimized out>) at gui/q_application_wrapper.cpp:35
#9  0x00007ffff6f6d3ca in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /nix/store/g6npkwh6m3rgl9hvc1j7vjg65jigrhc1-qtbase-5.15.2/lib/libQt5Core.so.5
#10 0x00007ffff73bd0e4 in QShortcutMap::dispatchEvent(QKeyEvent*) () from /nix/store/g6npkwh6m3rgl9hvc1j7vjg65jigrhc1-qtbase-5.15.2/lib/libQt5Gui.so.5
#11 0x00007ffff73bdceb in QShortcutMap::tryShortcut(QKeyEvent*) () from /nix/store/g6npkwh6m3rgl9hvc1j7vjg65jigrhc1-qtbase-5.15.2/lib/libQt5Gui.so.5
#12 0x00007ffff7366c0e in QWindowSystemInterface::handleShortcutEvent(QWindow*, unsigned long, int, QFlags<Qt::KeyboardModifier>, unsigned int, unsigned int, unsigned int, QString const&, bool, unsigned short) ()
   from /nix/store/g6npkwh6m3rgl9hvc1j7vjg65jigrhc1-qtbase-5.15.2/lib/libQt5Gui.so.5
#13 0x00007ffff7389b0b in QGuiApplicationPrivate::processKeyEvent(QWindowSystemInterfacePrivate::KeyEvent*) () from /nix/store/g6npkwh6m3rgl9hvc1j7vjg65jigrhc1-qtbase-5.15.2/lib/libQt5Gui.so.5
#14 0x00007ffff73635cc in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /nix/store/g6npkwh6m3rgl9hvc1j7vjg65jigrhc1-qtbase-5.15.2/lib/libQt5Gui.so.5
#15 0x00007fffe603263a in ?? () from /nix/store/mx4457i8a2sbpmhsbnbd1q0qd16v12zg-qtbase-5.15.2-bin/lib/qt-5.15.2/plugins/platforms/../../../../../g6npkwh6m3rgl9hvc1j7vjg65jigrhc1-qtbase-5.15.2/lib/libQt5XcbQpa.so.5
#16 0x00007ffff42125ab in g_main_context_dispatch () from /nix/store/dy0lmb10a75j2xvjhwkf5dkxa57qy1gr-glib-2.68.2/lib/libglib-2.0.so.0
#17 0x00007ffff4212858 in g_main_context_iterate.constprop () from /nix/store/dy0lmb10a75j2xvjhwkf5dkxa57qy1gr-glib-2.68.2/lib/libglib-2.0.so.0
#18 0x00007ffff421290f in g_main_context_iteration () from /nix/store/dy0lmb10a75j2xvjhwkf5dkxa57qy1gr-glib-2.68.2/lib/libglib-2.0.so.0
#19 0x00007ffff6fc5520 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /nix/store/g6npkwh6m3rgl9hvc1j7vjg65jigrhc1-qtbase-5.15.2/lib/libQt5Core.so.5
#20 0x00007ffff6f6bd7b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /nix/store/g6npkwh6m3rgl9hvc1j7vjg65jigrhc1-qtbase-5.15.2/lib/libQt5Core.so.5
#21 0x00007ffff6f74360 in QCoreApplication::exec() () from /nix/store/g6npkwh6m3rgl9hvc1j7vjg65jigrhc1-qtbase-5.15.2/lib/libQt5Core.so.5
#22 0x0000000000453dbb in main (argc=<optimized out>, argv=<optimized out>) at main.cpp:67
```

**Comment by rerrahkr** [comment]
Thank you for your report. #259 was not good enough yet, so I fixed it at 22d043a96524e541523d9795b1ea48fb90ba019b.
Note that track visibility settings is moved to Song tab in the latest dev-build.

**Comment by OPNA2608** [comment]
Closing because it's fixed on my Linux system and doesn't seem Windows specific. Let us know if this is somehow still a problem after 22d043a96524e541523d9795b1ea48fb90ba019b.
